### PR TITLE
Move BPF functionalities to utils

### PR DESCRIPTION
**Pull request by pxgao**
Some functions in core/modules/bpf.h(cc) are moved to
core/utils/bpf.h so that other modules can share these
functions.

### CODE REVIEW

File: core/modules/bpf.h [hunk 1]
```diff
@@ -37,8 +37,8 @@
 
 #include "../module.h"
 #include "../pb/module_msg.pb.h"
-
-using bpf_filter_func_t = u_int (*)(u_char *, u_int, u_int);
+#include "../utils/bpf.h"
+#include "pb/bpf_hook_msg.pb.h"
```
Comment (sangjinhan) [inline core/modules/bpf.h]: This file doesn't exist

File: core/modules/bpf.h [hunk 2]
```diff
@@ -57,23 +57,11 @@ class BPF final : public Module {
...
-#else
-    bpf_program il_code;
-#endif
-    int gate;
-    int priority;     // higher number == higher priority
-    std::string exp;  // original filter expression string
-  };
-
-  static bool Match(const Filter &, u_char *, u_int, u_int);
+   static bool Match(const bess::utils::Filter &, u_char *, u_int, u_int);
```
Comment (sangjinhan) [inline core/modules/bpf.h]: seems all lines in the private section are one whitespace off

File: core/utils/bpf.h [hunk 3]
```diff
@@ -0,0 +1,1108 @@
+// Copyright (c) 2017, Cloudigo.
```
Comment (sangjinhan) [inline core/utils/bpf.h]: not Cloudigo.

Comment (sangjinhan) [reply core/utils/bpf.h]: You should keep the original copyright lines
```
 /* x86_64 BPF JIT code was adopted from FreeBSD 10 - Sangjin */
 	 
 /*-
  * Copyright (C) 2002-2003 NetGroup, Politecnico di Torino (Italy)
  * Copyright (C) 2005-2009 Jung-uk Kim <jkim@FreeBSD.org>
  * All rights reserved.
...
```

**Comment by pxgao** [comment]
@sangjinhan Made the changes.

**Comment by sangjinhan** [comment]
Merged, thanks!

ha6zectleg4sfljwhqf5jm4so1_400

**Pull request merged**